### PR TITLE
Improve testing

### PR DESCRIPTION
**Pull request by jsvd**
* replace integration tests with unit tests
* move testing to docker to make them easier to setup and improve speed

- [x] TODO move versions url to elastic/logstash

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+services: docker
+addons:
+  apt:
+    packages:
+      - docker-ce
+matrix:
+  include:
+  - env: ELASTIC_STACK_VERSION=5.6.13
+  - env: ELASTIC_STACK_VERSION=6.4.3
+  - env: ELASTIC_STACK_VERSION=7.0.0-alpha1-SNAPSHOT
```
Comment (colinsurprenant) [inline .travis.yml]: I know I've asked this question in another PR but hardcoding versions will not scale, when the stack is upgraded we will have to update all plugins `.travis.yml`. Isn't there  a way for us to dynamically inject a matrix somehow which can be maintained and updated externally?

**Comment by jsvd** [comment]
@colinsurprenant let me know what do you think now, put the version dictionary in a gist url and now travis fetches that to figure out with patch version to download for a given minor

**Comment by jsvd** [comment]
@colinsurprenant let me know what do you think now. Version file contains:
```json
{
  "releases": {
    "5.x": "5.6.13",
    "6.x": "6.5.1",
    "7.x": "7.0.0-alpha1"
  },
  "snapshots": {
    "6.x": "6.6.0-SNAPSHOT",
    "7.x": "7.0.0-SNAPSHOT"
  }
}
```

File: ci/unit/docker-setup.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    fi
+    if [[ "$ELASTIC_STACK_RETRIEVED_VERSION" != "null" ]]; then
+      # remove starting and trailing double quotes
+      ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION%\"}"
+      ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION#\"}"
+      echo "Translated $ELASTIC_STACK_VERSION to ${ELASTIC_STACK_RETRIEVED_VERSION}"
+      export ELASTIC_STACK_VERSION=$ELASTIC_STACK_RETRIEVED_VERSION
+    fi
+
+    echo "Testing against version: $ELASTIC_STACK_VERSION"
```
Comment (colinsurprenant) [inline ci/unit/docker-setup.sh]: what happens here if the `$ELASTIC_STACK_RETRIEVED_VERSION` retrieval/parsing failed? the `travis.yml` value like `6.x` will not be valid?

Comment (colinsurprenant) [reply ci/unit/docker-setup.sh]: I mean, will not be valid propagted downstream into `Dockerfile` ?

**Comment by jsvd** [comment]
failures:

```
# when internet is down
% SNAPSHOT=true ELASTIC_STACK_VERSION=6.x ./ci/unit/docker-setup.sh && ./ci/unit/docker-run.sh | jq
Fetching versions from https://gist.githubusercontent.com/jsvd/12c60459ba0cc505dc56867561b41806/raw/f7ab015f496434cad1e97b9e4ccff0f00c31344a/versions.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: gist.githubusercontent.com
% echo $?
6
```

```
# when the file isn't proper JSON
% SNAPSHOT=true ELASTIC_STACK_VERSION=6.x ./ci/unit/docker-setup.sh && ./ci/unit/docker-run.sh | jq
Fetching versions from https://gist.githubusercontent.com/jsvd/36c760a2ec4e67831dd52cac518f088a/raw/198615a2ac02ac5d039333bafdc70662d7dba7a2/gen.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2193  100  2193    0     0   1145      0  0:00:01  0:00:01 --:--:--  1145
parse error: Invalid numeric literal at line 1, column 2
% echo $?
4
```

**Pull request merged**